src/libAtomVM/context.c

```diff
@@ -107,8 +107,16 @@ Context *context_new(GlobalContext *glb)
 
 void context_destroy(Context *ctx)
 {
+    struct ListHead *dummy = synclist_wrlock(&ctx->global->processes_table);
+    UNUSED(dummy);
+
     // Another process can get an access to our mailbox until this point.
-    synclist_remove(&ctx->global->processes_table, &ctx->processes_table_head);
+    list_remove(&ctx->processes_table_head);
+
+    // Ensure process is not registered
+    globalcontext_maybe_unregister_process_id(ctx->global, ctx->process_id);
```

Comment:
I'm not sure why we need to keep a lock while we are unregistering it.

Reply:
You are correct. It is the `registered_processes` that I need to lock temporarily when the context begins to go down, until after it is unregistered. Otherwise there could be a race, another process could `whereis` the pid of the process that was dying, but not unregistered yet.
I beleive this means `globalcontext_maybe_unregister_process_id` should be `globalcontext_unregister_process_id_nolock` - removing the use of `synclist_wrlock`, which should be used at the beginning of `context_destroy` and unlocked after `globalcontext_unregister_process_id_nolock`.

Reply:
It appears I was wrong about a race condition. After some extensive testing I could not find a statistical difference with whereis returning more bad pids either with or without a rdlock on `registered_processes`.